### ISSUE TITLE
Backfill CRIS data 

### ISSUE THREAD

**Issue by johnclary** [cityofaustin/atd-data-tech#16824]
Since we [fixed the CRIS import](https://github.com/cityofaustin/atd-data-tech/issues/16823), we are missing a few days' worth of crash updates. We need to backfill the DB with crashes processed since ~4/12. The last successful extract we processed was dated 4/13/2024, but we are not 100% sure what date range that extract covered.

**Comment by johnclary** [comment]
This is blocked until we regain Chia's CRIS access. 

@frankhereford is going to prep our bastion host for running the CRIS import backfill.
@mddilley will be on point for running the CRIS import from the bastion host once we regain CRIS access.

**Comment by frankhereford** [comment]
## TLDR

Here are some instructions on using the bastion as a host to run the CRIS import in dev mode.

## Intent 

I got us setup with a shared checkout on the bastion. I've done it so that we don't have to use the root account to use it. This is accomplished using groups, since we're all members of the `wheel` group. This is one of those invisible, best practices type thing, since we never invoke root's permissions (except for using docker.) Anyway - that's neither here nor there. You can:

## Instructions

```bash
ssh <username>@rds-bastion.austinmobility.io;
cd /opt/vision-zero/atd-vz-data/atd-etl/cris_import/development_extracts;

# copy in the extract zip archive into this folder
wormhole receive <whatever extract you want>; # or use scp or what-have-you. wormhole is available.

cd ..;
docker compose run cris-import;
```

## special stuff for this install

I have hard-code a single configuration in the ETL because we are **on** the bastion host, and this ETL uses the bastion as a `ssh` relay.  You can see the change in the un-checked-in changes with `git diff`. 

I have set the environment to be `production` so it connects on to the right DB instance and correct graphql-engine endpoints.

I've tested that it can connect to the DB and that it can access secrets. I've put the IP in the graphql-engine security group that it needs, but I have not made any writing changes to the DB, so I don't have a great way to test that. Heads up if it can't insert things into the conflict resolution system .. I would look here.

cc/ @johnclary @mddilley @chiaberry

**Comment by mddilley** [comment]
thanks, Frank! I ssh'd to the bastion and tested everything short of transferring files and running the script. All works great! 🙏

**Comment by mddilley** [comment]
for the full story see [this thread](https://austininnovation.slack.com/archives/CM9MK950S/p1713359994138409)

in summary:
- Frank set up the recurring daily CRIS extracts from now until the end of the year so we are back in action there
- this morning, we received two pdfs on the SFTP server: 
   - one retroactive extract containing records processed between the start of the year and today
      - this was discarded and not imported
   - one extract containing the usual daily extract
      - this extract was imported successfully on the bastion host stack
- Frank also requested a backfill extract going back to 4/12 as described in this issue description
   - this one initially failed on the bastion host, but I ran it locally and it went through

Next steps are to run the threaded CR3 downloader (in progress), run the thread CR3 metadata script (todo), and figure out what to do to catch up on OCR.

**Comment by mddilley** [comment]
I'm keeping this in progress while we figure out what is going on with the CRIS extract deliveries.

Slack threads:
[4/18/2024](https://austininnovation.slack.com/archives/CM9MK950S/p1713446596585419)
[4/19/2024](https://austininnovation.slack.com/archives/CM9MK950S/p1713533603680489)

In summary, the SFTP delivery pipeline seems broken so I'm requesting extracts using the **Download** option. We can download these locally and then either copy them to the SFTP folder and trigger the DAG **or** run them locally. I'm going to go the DAG route so we capture logs.

**Comment by mddilley** [comment]
@johnclary To close this one out, I wrote up some [basic docs about the CRIS request process](https://app.gitbook.com/o/-LzDQOVGhTudbKRDGpUA/s/-M4Ul-hSBiM-3KkOynqS/vision-zero-crash-database/visionzero-sftp-etl#how-do-these-files-get-extracted), and @frankhereford gave some great feedback on how to transfer files to the SFTP when we need to run manually. Here is [the thread](https://austininnovation.slack.com/archives/CM9MK950S/p1713889462993449), and it sounds like you've already been using this method. I like the idea a lot!
